### ISSUE TITLE
GitHub is sending us PR synchronize event for a PR that does not exist.

### ISSUE THREAD

**Issue by Mariatta** [python/bedevere#109]
Same issue as described in https://github.com/python/the-knights-who-say-ni/issues/148.
Bedevere is attempting to apply a label to such PR, causing `gidgethub.BadRequest: Not Found` error.

https://github.com/python/bedevere/blob/7cd16efa18fa0bd2c902fea8732623d8d54169bc/bedevere/bpo.py#L35-L42

Perhaps we can catch the error here, so I won't keep getting error alerts.
Thanks.

**Comment by brettcannon** [comment]
Yep, if GitHub gives us bad data then we should log that fact and then move on.

**Comment by Mariatta** [comment]
🍝 (copy pasta 😄)

Update from GitHub:

PR 5581 exists, but it just has been marked as "suspicious/spammy", so we and the bots can't access it. The PR was still open and we were receiving the "synchronize" events, breaking the bots.

I have asked GitHub to close the PR, and it has been closed now.

We can still improve the code though if this ever happens again :)

**Comment by lysnikolaou** [comment]
I'd like to work on this. Do you think that every call to `post` should be wrapped in a `try/except` clause? If that's the case, maybe we could use audit hooks as well.

**Comment by brettcannon** [comment]
@lysnikolaou this definitely isn't a priority to fix, but yes, a well-documents `try`/`except` with a tight `except` clause would be the way to handle it.
